deploy-domains-env/action.yml

```diff
@@ -0,0 +1,79 @@
+name: Deploy Domains Infrastructure
+description: Deploy Domains for infrastructure
+
+inputs:
+  azure-credentials:
+    description: A JSON string containing service principal credentials.
+    required: true
+  environment:
+    description: the name of the environment for the domains.
+    required: true
+  slack-webhook:
+    description: Name of the slack webhook
+    required: false
+  healthcheck:
+    description: Health check path
+    required: false
+  terraform-base:
+    description: Path to the terraform files
+    required: false
+    default: "terraform/domains/environment_domains"
+
+runs:
+  using: composite
+  steps:
+    - name: Checkout code
+      uses: actions/checkout@v4
+
+    - name: Domains Changes
+      uses: dorny/paths-filter@v2
+      id: domains_changes
+      with:
+        filters: |
+          directory:
+            - '${{ inputs.terraform-base }}/**/*'
+
+    - name: Set Environment variables
+      id: set_env_var
+      if: steps.domains_changes.outputs.directory == 'true'
+      shell: bash
+      run: |
+        terraform_version=$(awk '/{/{f=/^terraform/;next}f' ${{ inputs.terraform-base }}/terraform.tf | grep -o [0-9\.]*)
+        echo "TERRAFORM_VERSION=$terraform_version" >> $GITHUB_ENV
+
+    - name: Use Terraform ${{ env.TERRAFORM_VERSION }}
+      if: steps.domains_changes.outputs.directory == 'true'
+      uses: hashicorp/setup-terraform@v3
+      with:
+        terraform_version: ${{ env.TERRAFORM_VERSION }}
+        terraform_wrapper: false
+
+    - uses: azure/login@v2
+      if: steps.domains_changes.outputs.directory == 'true'
+      with:
+        creds: ${{ inputs.azure-credentials }}
+
+    - name: Terraform Domains Environment
+      if: steps.domains_changes.outputs.directory == 'true'
+      shell: bash
+      run: |
+        make ci ${{ inputs.environment }} domains-plan
+
+    - name: Run healthcheck
+      if: steps.domains_changes.outputs.directory == 'true' && ${{ inputs.healthcheck != '' }}
+      shell: bash
+      run: |
+        origin_hostnames=$(terraform -chdir=${{ inputs.terraform-base }} output -json origin_hostnames | jq -r '.[]')
```

Comment:
origin_hostname will be teacherservices.cloud. Can we get the gov.uk url instead?